s3-handler/main.go

```diff
@@ -131,13 +151,55 @@ func Handler(request events.S3Event) (Response, error) {
 	}, nil
 }
 
+func parseSampleRateRules(input []byte) ([]sampleRateRule, error) {
+	var result []sampleRateRule
+	err := json.Unmarshal(input, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, rule := range result {
+		if rule.SampleRate == 0 {
+			rule.SampleRate = 1
+		}
+	}
+
+	return result, nil
+}
+
+func initSamplingRules() error {
+	rulesRaw := os.Getenv("SAMPLE_RATE_RULES")
+	logrus.WithField("SAMPLE_RATE_RULES", rulesRaw).Info("Got sample rate rules")
+
+	if rulesRaw == "" {
+		sampleRateRules = make([]sampleRateRule, 0)
+		return nil
+	}
```

Comment:
Why are you setting this to an empty slice? Don't think this is necessary.

Reply:
The zero-value of a `[]T` is `nil` but `len`, `append`, and `range` all allow passing a `nil` and doing the right thing without exploding, so it's usually fine to leave it uninitialized.